apps/api/pages/api/destination-calendars/_post.ts

```diff
@@ -68,17 +67,36 @@ async function postHandler(req: NextApiRequest) {
   const assignedUserId = isAdmin ? parsedBody.userId || userId : userId;
 
   /* Check if credentialId data matches the ownership and integration passed in */
-  const credential = await prisma.credential.findFirst({
-    where: { type: parsedBody.integration, userId: assignedUserId },
-    select: { id: true, type: true, userId: true },
+  const userCredentials = await prisma.credential.findMany({
+    where: {
+      type: parsedBody.integration,
+      userId: assignedUserId,
+    },
+    select: credentialForCalendarServiceSelect,
   });
 
-  if (!credential)
+  if (!userCredentials)
+    throw new HttpError({
+      statusCode: 400,
+      message: "Bad request, credential id invalid",
+    });
+
+  const calendarCredentials = getCalendarCredentials(userCredentials);
+
+  const { connectedCalendars } = await getConnectedCalendars(calendarCredentials, [], parsedBody.externalId);
+  const filteredCalendars = connectedCalendars.filter(
+    (c) =>
+      c?.primary?.externalId === parsedBody.externalId && c?.primary?.integration === parsedBody.integration
+  );
+
+  if (filteredCalendars.length === 0)
     throw new HttpError({
       statusCode: 400,
       message: "Bad request, credential id invalid",
     });
 
+  const credentialId = filteredCalendars[0].primary?.credentialId;
+
```

Comment:
We get connected calendars for this user and fetch credentialId internally, without exposing it.